From 31d7560e77727113eceea460c93f574aa8015c25 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:34 -0400
Subject: [tools] perf: Fix 'make help' message error

Message-id: <1410181752-25631-270-git-send-email-jolsa@redhat.com>
Patchwork-id: 91947
O-Subject: [PATCH RHEL7.1 BZ1134356 269/307] perf tools: Fix 'make help' message error
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit fc9cabeabf42d76854059e7bce81a02645e7e5ca
Author: Jianyu Zhan <nasa4836@gmail.com>
Date: Tue Jun 3 00:44:34 2014 +0800

description
===========
Currently 'make help' message has such hint:

   use "make prefix=<path> <install target>" to install to a particular
       path like make prefix=/usr/local install install-doc

But this is misleading, when I specify "prefix=/usr/local", it has got no
respect at all.

This is because that, "DESTDIR" is considered first. In this case, "DESTDIR"
has an empty value, so "prefix" is honored. However, "prefix" is unconditionally
assigned to $HOME, regardless of what it is set to from command line. So our
"prefix" setting got no respect and the actual destination falls back to $HOME.

This patch fixes this issue and corrects the help message.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Makefile.perf b/tools/perf/Makefile.perf
index 6286e13..ae20edf 100644
--- a/tools/perf/Makefile.perf
+++ b/tools/perf/Makefile.perf
@@ -789,8 +789,8 @@ help:
  @echo ''
  @echo 'Perf install targets:'
  @echo '  NOTE: documentation build requires asciidoc, xmlto packages to be installed'
- @echo '  HINT: use "make prefix=<path> <install target>" to install to a particular'
- @echo '        path like make prefix=/usr/local install install-doc'
+ @echo '  HINT: use "prefix" or "DESTDIR" to install to a particular'
+ @echo '        path like "make prefix=/usr/local install install-doc"'
  @echo '  install - install compiled binaries'
  @echo '  install-doc - install *all* documentation'
  @echo '  install-man - install manpage documentation'
diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 319426f..4f100b5 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -600,7 +600,7 @@ endif
 
 # Make the path relative to DESTDIR, not to prefix
 ifndef DESTDIR
-prefix = $(HOME)
+prefix ?= $(HOME)
 endif
 bindir_relative = bin
 bindir = $(prefix)/$(bindir_relative)
-- 
1.7.1